src/components/ReportViewer/log-report.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import LogReportItem from './log-report-item';
+import { v4 as uuid4 } from 'uuid';
+
+// WARNING this file has been copied from commons-ui, and updated here. Putting it back to commons-ui has to be discussed.
+
+/**
+ * A LogReport is a tree that can contain 3 kinds of nodes:
+ * - GlobalReport : an optional top-level node that contains only the NodeReport children
+ * - NodeReport : contain a root node report having N SubReport children
+ * - SubReport : contain other SubReport children, and can have some reports (log messages displayed in the right pane)
+ */
+export const LogReportType = {
+    GlobalReport: 'global report',
+    NodeReport: 'node report',
+    SubReport: 'reporter',
+};
+
+export default class LogReport {
+    constructor(reportType, jsonReporter, parentReportId) {
+        this.id = undefined;
```

Comment:
Really necessary to initilialize nodeId wtih defferent ways
Perharps just  this.`nodeId = uuid4()`

Rename by `uniqueId`

Reply:
yes done